main.nf

```diff
@@ -104,81 +126,201 @@ params.gff = params.genome ? params.genomes[ params.genome ].gff ?: false : fals
+Channel.fromPath("$baseDir/assets/where_are_my_files.txt", checkIfExists: true)
        .into{ch_where_trim_galore; ch_where_star; ch_where_hisat2; ch_where_hisat2_sort}
 
 // Define regular variables so that they can be overwritten
 clip_r1 = params.clip_r1
 clip_r2 = params.clip_r2
 three_prime_clip_r1 = params.three_prime_clip_r1
 three_prime_clip_r2 = params.three_prime_clip_r2
-forward_stranded = params.forward_stranded
-reverse_stranded = params.reverse_stranded
-unstranded = params.unstranded
+forwardStranded = params.forwardStranded
+reverseStranded = params.reverseStranded
+unStranded = params.unStranded
 
 // Preset trimming options
 if (params.pico){
     clip_r1 = 3
     clip_r2 = 0
     three_prime_clip_r1 = 0
     three_prime_clip_r2 = 3
-    forward_stranded = true
-    reverse_stranded = false
-    unstranded = false
+    forwardStranded = true
+    reverseStranded = false
+    unStranded = false
+}
+
+// Get rRNA databases
+def returnFile(it) {
+    inputFile = file(it)
+    if (!file(inputFile).exists()) exit 1, "File ${inputFile} does not exist!"
+    return inputFile
+}
+if(params.rRNA_database_manifest){
+    rRNA_database = returnFile(params.rRNA_database_manifest)
+    if (rRNA_database.isEmpty()) {exit 1, "File ${rRNA_database.getName()} is empty!"}
+    Channel
+        .from( rRNA_database.readLines() )
+        .map { row -> returnFile(row) }
+        .set { sortmerna_fasta }
+} else {
+    Channel.fromPath([
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/rfam-5.8s-database-id98.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/rfam-5s-database-id98.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-arc-16s-id95.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-arc-23s-id98.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-bac-16s-id90.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-bac-23s-id98.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-euk-18s-id95.fasta',
+        'https://raw.githubusercontent.com/biocore/sortmerna/master/rRNA_databases/silva-euk-28s-id98.fasta'])
+        .set { sortmerna_fasta }
 }
 
 // Validate inputs
 if (params.aligner != 'star' && params.aligner != 'hisat2'){
     exit 1, "Invalid aligner option: ${params.aligner}. Valid options: 'star', 'hisat2'"
 }
-if( params.star_index && params.aligner == 'star' ){
+if (params.pseudo_aligner && params.pseudo_aligner != 'salmon'){
+    exit 1, "Invalid pseudo aligner option: ${params.pseudo_aligner}. Valid options: 'salmon'"
+}
+
+
+if( params.star_index && params.aligner == 'star' && !params.skipAlignment ){
+  if (params.compressedReference){
```

Comment:
Do with need `--compressedReference` option? Can we not just inspect the given files to see if they have a `.gz` file extension?

Reply:
We auto-detect this now 👍

Reply:
@olgabot - I think that if they do that, then they deserve for their pipeline run to crash 😰 

But yeah, I think more people are likely to be caught out by not giving `--compressedReference` than being caught out by having weird filenames.